### PR TITLE
Add playerctld support to music block

### PR DESCRIPTION
**Pull request by bim9262**
playerctld registers itself like any other mpris media player so I wanted to make sure it wasn't showing up on the list.

I ended up making it a default `interface_name_exclude`, but perhaps it would be better to hard code in as interface that should be ignored so that if a user sets `interface_name_exclude` playerctld is still ignored.  Let me know if you have any thoughts on that.

### CODE REVIEW

**Comment by bim9262** [comment]
Perhaps something like:

```diff
diff --git a/src/blocks/music.rs b/src/blocks/music.rs
index 524c7071..972f9478 100644
--- a/src/blocks/music.rs
+++ b/src/blocks/music.rs
@@ -20,7 +20,7 @@                                                                                                                                                                                                                                             b-23tower" 19:16 01-Feb-23
 //! ----|--------|--------
 //! `format` | A string to customise the output of this block. See below for available placeholders. | <code>" $icon {$combo.rot-str() $play &vert;}"</code>
 //! `player` | Name(s) of the music player(s) MPRIS interface. This can be either a music player name or an array of music player names. Run <code>busctl --user list &vert; grep "org.mpris.MediaPlayer2." &vert; cut -d' ' -f1</code> and the name is the part after "org.mpris.MediaPlayer2.". | `None`
-//! `interface_name_exclude` | A list of regex patterns for player MPRIS interface names to ignore. | `["playerctld"]`
+//! `interface_name_exclude` | A list of regex patterns for player MPRIS interface names to ignore. | `[]`
 //! `separator` | String to insert between artist and title. | `" - "`
 //! `seek_step` | Number of microseconds to seek forward/backward when scrolling on the bar. | `1000`
 //!
@@ -117,7 +117,6 @@ const PREV_BTN: &str = "prev_btn";
 pub struct Config {
     format: FormatConfig,
     player: PlayerName,
-    #[default(vec!["playerctld".into()])]
     interface_name_exclude: Vec<String>,
     #[default(" - ".into())]
     separator: String,
@@ -165,8 +164,11 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
         PlayerName::Single(name) => vec![name],
         PlayerName::Multiple(names) => names,
     };
-    let exclude_regex = config
-        .interface_name_exclude
+
+    let mut interface_name_exclude = config.interface_name_exclude;
+    interface_name_exclude.push("playerctld".into());
+
+    let exclude_regex = interface_name_exclude
         .iter()
         .map(|r| Regex::new(r))
         .collect::<Result<Vec<_>, _>>()
```

**Comment by bim9262** [comment]
Another possible improvement whether you use playerctld or not, could be to switch the player to the media player that is actively being used.


```diff
diff --git a/src/blocks/music.rs b/src/blocks/music.rs
index 524c7071..c0d32c8d 100644
--- a/src/blocks/music.rs
+++ b/src/blocks/music.rs
@@ -292,7 +292,7 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
                     let args = msg.args().unwrap();
                     let header = msg.header().unwrap();
                     let sender = header.sender().unwrap().unwrap();
-                    if let Some(player) = players.iter_mut().find(|p| &*p.owner == sender) {
+                    if let Some((pos, player)) = players.iter_mut().enumerate().find(|p| &*p.1.owner == sender) {
                         let props = args.changed_properties;
                         if let Some(status) = props.get("PlaybackStatus") {
                             let status: &str = status.downcast_ref().unwrap();
@@ -302,6 +302,13 @@ pub async fn run(config: Config, mut api: CommonApi) -> Result<()> {
                             player.metadata =
                                 zbus_mpris::PlayerMetadata::try_from(metadata.to_owned()).unwrap();
                         }
+                        if player.status == Some(PlaybackStatus::Playing) {
+                            if players.is_empty() {
+                                cur_player = None;
+                            } else{
+                                cur_player = Some(pos);
+                            }
+                        }
                         break;
                     }
                 }
```

**Comment by MaxVerevkin** [comment]
> Another possible improvement whether you use playerctld or not, could be to switch the player to the media player that is actively being used.

This has been done in cef6e14ad796ebd36aff6290d6011305c667bc9d, but without the `if player.status == Some(PlaybackStatus::Playing)` check, which I suppose should be added.

**Comment by bim9262** [comment]
> This has been done in https://github.com/greshake/i3status-rust/commit/cef6e14ad796ebd36aff6290d6011305c667bc9d, but without the if player.status == Some(PlaybackStatus::Playing) check, which I suppose should be added.

I added in a check for `player.status == Some(PlaybackStatus::Playing)`.

File: files/icons/awesome.toml [hunk 1]
```diff
@@ -47,10 +47,10 @@ memory_mem = "\uf2db" # fa-microchip
 memory_swap = "\uf0a0" # fa-hdd-o
 mouse = "\uf245" # fa-mouse-pointer
 music = "\uf001" # fa-music
-music_next = "\uf061" # fa-arrow-right
+music_next = "\uf050" # fa-forward-fast
 music_pause = "\uf04c" # fa-pause
 music_play = "\uf04b" # fa-play
-music_prev = "\uf060" # fa-arrow-left
+music_prev = "\uf049" # fa-backward-fast
```
Comment (bim9262) [inline files/icons/awesome.toml:53]: I didn't mean to add these font changes in this commit, but now that I have... Any thoughts on swapping out the plain arrows for the media player style forward and back buttons?

I know I can just override them in my config, if I don't like the defaults, but I thought that these are more appropriate.

Comment (MaxVerevkin) [reply files/icons/awesome.toml:55]: Yeah, `fa-backward-step` / `fa-forward-step` might me the most appropriate. Can you open a new PR for that?

**Pull request merged**